R/npn_data_download.R

```diff
@@ -719,10 +731,22 @@ npn_get_data <- function(url,
                                promote_num_to_string = TRUE) %>%
       tibble::as_tibble() %>%
       #replace missing data indicator with NA
-      dplyr::mutate(dplyr::across(dplyr::where(is.numeric),
-                                  \(x) ifelse(x == -9999, NA_real_, x))) %>%
-      dplyr::mutate(dplyr::across(dplyr::where(is.character),
-                                  \(x) ifelse(x == "-9999", NA_character_, x)))
+      dplyr::mutate(
+        dplyr::across(dplyr::where(is.numeric),
+                      function(x) ifelse(x == -9999, NA_real_, x))
+      ) %>%
+      dplyr::mutate(
+        dplyr::across(dplyr::where(is.character),
+                      function(x) ifelse(x == "-9999", NA_character_, x))
+      ) %>%
+      #handle some columns that may be read in as the wrong type if they happen
+      #to be all NAs (#87). `any_of()` is used because not all endpoints will
+      #return these columns!
+      dplyr::mutate(
+        dplyr::across(dplyr::any_of("update_datetime"),
+                      function(x) as.POSIXct(x, tz = "UTC")),
+        dplyr::across(dplyr::any_of("intensity_value"), as.character)
```

Comment:
I've found that this column is sometimes numbers, but also sometimes things like "3 to 10".  Do you happen to know if `abundance_value` is the same?

Reply:
I just checked the Abundance Value table and some random examples are "50-74%", "Peak pollen", "Moderate".

Reply:
Ah ok,  I'll make a separate PR to make this change since this one is already merged
```suggestion
        dplyr::across(dplyr::any_of("intensity_value", "abundance_valuue"), as.character)
```